CMakeLists.txt

```diff
@@ -1032,21 +1032,8 @@ zephyr_compile_options(
   $<$<COMPILE_LANGUAGE:C>:-std=${CSTD}>
 )
 
-configure_file(
-     $ENV{ZEPHYR_BASE}/include/arch/common/app_data_alignment.ld
```

Comment:
I suspect, but cannot prove, that this code is badly written.

Files should not be copied into directories that are in the include path, instead the include path scope should be increased to include the source file.

If we were to rectify this, I suspect we would not need any portability layer, as the include path could be the same across toolchains.

The fewer toolchain abstractions the better.

Reply:
It may very well be badly written, but how then do we rectify it?
If the solution doesn't require too much effort, I'll volunteer to do it with your guidance.

Otherwise, I'd default to just moving it like this PR does.

Reply:
I can't find an easy way to do it, and since this is a pre-existing problem, this is OK.